addons/pause/check-online.js

```diff
@@ -0,0 +1,90 @@
+let hasOnlineFeatures;
+
+/** Guesses whether the project has cloud variable features */
+const scanForOnlineFeatures = (vm, console, msg, element) => {
+  // Well, only if it actually has cloud variables :P
+  if (!vm.runtime.hasCloudData()) {
+    hasOnlineFeatures = false;
+    return;
+  }
+
+  const allBlocks = vm.runtime.targets.flatMap((target) => Object.values(target.blocks._blocks));
+  // Map the names and values of cloud variables to an object
+  const cloudVariables = {};
+  const globalVariables = vm.runtime.getTargetForStage().variables;
+  for (let id of Object.keys(globalVariables)) {
+    if (globalVariables[id].isCloud) {
+      cloudVariables[globalVariables[id].name] = globalVariables[id].value;
+    }
+  }
+
+  const names = Object.keys(cloudVariables);
+  const values = Object.values(cloudVariables);
+
+  const varsHaveInvalidNumbers = values.some((i) => String(Number(i)) !== i);
```

Comment:
this condition might trigger false positives if a cloud variable's value is an actual number? (because a string is never strictly equal to a number). not sure if it's intentional or not though